src/pages/workspace/categories/ImportedCategoriesPage.tsx

```diff
@@ -0,0 +1,150 @@
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+
+type ImportedCategoriesPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.CATEGORIES_IMPORTED>;
+
+function numberToColumn(index: number) {
+    let column = '';
+    let number = index;
+    while (number >= 0) {
+        column = String.fromCharCode((number % 26) + 65) + column;
+        number = Math.floor(number / 26) - 1;
+    }
+    return column;
+}
+
+function generateColumnNames(length: number) {
+    return Array.from({length}, (_, i) => numberToColumn(i));
+}
+
+function findDuplicate(array: string[]): string | null {
+    const frequencyCounter: Record<string, number> = {};
+
+    for (const item of array) {
+        if (item !== CONST.CSV_IMPORT_COLUMNS.IGNORE) {
+            if (frequencyCounter[item]) {
+                return item;
+            }
+            frequencyCounter[item] = (frequencyCounter[item] || 0) + 1;
+        }
+    }
+
+    return null;
+}
+
+function ImportedCategoriesPage({route}: ImportedCategoriesPageProps) {
+    const {translate} = useLocalize();
+    const [spreadsheet] = useOnyx(ONYXKEYS.IMPORTED_SPREADSHEET);
+    const [isImportingCategories, setIsImportingCategories] = useState(false);
+    const [containsHeader, setContainsHeader] = useState(false);
+    const policyID = route.params.policyID;
+    const columnNames = generateColumnNames(spreadsheet?.data?.length ?? 0);
+
+    const columnRoles: ColumnRole[] = [
+        {text: translate('common.ignore'), value: CONST.CSV_IMPORT_COLUMNS.IGNORE},
+        {text: translate('common.name'), value: CONST.CSV_IMPORT_COLUMNS.NAME, isRequired: true},
+        {text: translate('common.enabled'), value: CONST.CSV_IMPORT_COLUMNS.ENABLED, isRequired: true},
+    ];
+
+    const requiredColumns = columnRoles.filter((role) => role.isRequired).map((role) => role);
+
+    const validate = useCallback(() => {
+        const columns = Object.values(spreadsheet?.columns ?? {});
+        let errors: Record<string, string | null> = {};
+
+        if (!requiredColumns.every((requiredColumn) => columns.includes(requiredColumn.value))) {
+            // eslint-disable-next-line rulesdir/prefer-early-return
+            requiredColumns.forEach((requiredColumn) => {
+                if (!columns.includes(requiredColumn.value)) {
+                    errors.required = translate('spreadsheet.fieldNotMapped', requiredColumn.text);
+                }
+            });
+        } else {
+            const duplicate = findDuplicate(columns);
+            if (duplicate) {
+                errors.duplicates = translate('spreadsheet.singleFieldMultipleColumns', duplicate);
+            } else {
+                errors = {};
+            }
+        }
+
+        return errors;
+    }, [requiredColumns, spreadsheet?.columns, translate]);
+
+    const importCategories = useCallback(() => {
+        validate();
+        const columns = Object.values(spreadsheet?.columns ?? {});
```

Comment:
I think we are missing GL Code from the columns if it is a Collect policy.

Reply:
You mean Control policy right? In Collect policy there is no way to add GL Code

Reply:
Added ✅